File: data_validation/scripts/r/validate_data.R

df <- read.csv('../sample_data/sample_data.csv')

# Missing values
cat("Missing values:\n")
print(colSums(is.na(df)))

# Duplicates
cat("\nDuplicate IDs:\n")
print(df[duplicated(df$id), ])

# Invalid gender
cat("\nInvalid gender values:\n")
print(subset(df, !(gender %in% c('M', 'F'))))

File: data_validation/scripts/python/validate_data.py

import pandas as pd

# Load data
df = pd.read_csv('../sample_data/sample_data.csv')

# Check missing values
print("Missing values:")
print(df.isnull().sum())

# Check duplicates by ID
print("\nDuplicate rows (by ID):")
print(df[df.duplicated('id')])

# Check invalid gender values
print("\nInvalid gender values (allowed: M, F):")
print(df[~df['gender'].isin(['M', 'F'])])
